catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/CollaborationMetacardPolicyOptimizationPlugin.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.security;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.Subject;
+import ddf.security.SubjectIdentity;
+import ddf.security.SubjectUtils;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.shiro.SecurityUtils;
+import org.geotools.filter.visitor.DefaultFilterVisitor;
+import org.opengis.filter.PropertyIsEqualTo;
+import org.opengis.filter.PropertyIsLike;
+
+/**
+ * Plugin that looks for queries that target collaboration metacards (workspaces, forms, etc) and
+ * updates the query filter so part of the collaboration metacard security policy is baked in,
+ * removing a significant burden from post-query filtering.
+ *
+ * <p>When identifying a {@link org.opengis.filter.Filter} that only selects collaboration
+ * metacards, the current implementation takes a best effort approach on the entire filter as a
+ * whole. It does not perform modification on individual subtrees, only the entire filter, or not at
+ * all. This leads to some behavioral notes:
+ *
+ * <ul>
+ *   <li>The presence of any metacard tag that does not imply identity as a collaboration metacard
+ *       will exclude the entire filter from being modified with policy information because it is
+ *       not a safe assumption.
+ *   <li>The lack of any metacard tag predicate on the filter will exclude the entire filter from
+ *       being modified with policy information because it is not a safe assumption.
+ * </ul>
+ */
+public class CollaborationMetacardPolicyOptimizationPlugin implements PreQueryPlugin {
+
+  private final SubjectIdentity identity;
+
+  public CollaborationMetacardPolicyOptimizationPlugin(SubjectIdentity identity) {
+    this.identity = identity;
+  }
+
+  @Override
+  public QueryRequest process(QueryRequest input)
+      throws PluginExecutionException, StopProcessingException {
+    return null;
```

Comment:
shouldn't we just return `input` here instead of wiping out the query request?

Reply:
Yeah I only filled out the test harness; the impl is still empty.